rewrite-pattern.js

```diff
@@ -728,9 +809,51 @@ const rewritePattern = (pattern, flags, options) => {
 	};
 
 	const tree = parse(pattern, flags, regjsparserFeatures);
+
+	if (config.transform.modifiers) {
+		const allDisabledModifiers = Object.create(null)
+		const itemStack = [tree];
+		let node;
+		while (node = itemStack.pop(), node != undefined) {
+			if (Array.isArray(node)) {
+				Array.prototype.push.apply(itemStack, node);
+			} else if (typeof node == 'object' && node != null) {
+				for (const key of Object.keys(node)) {
+					const value = node[key];
+					if (key == "modifierFlags") {
+						if (value.disabling.length > 0){
+							value.disabling.split("").forEach((flag)=>{
+								allDisabledModifiers[flag] = true
+							});
+						}
+					} else if (typeof value == 'object' && value != null) {
+						itemStack.push(value);
+					}
+				}
+			}
+		}
+
+		for (const flag of Object.keys(allDisabledModifiers)) {
+			config.modifiersData[flag] = true;
+		}
+	}
+
 	// Note: `processTerm` mutates `tree` and `groups`.
 	processTerm(tree, regenerateOptions, groups);
 	assertNoUnmatchedReferences(groups);
+
+	const onNewFlags = options && options.onNewFlags;
+	if (onNewFlags) onNewFlags(flags.split("").filter((flag) => {
```

Comment:
I really dislike using a callback for this, since (differently from `onNamedGroup`) it's only called once. It feels like those C++ "return pattern" of passing pointers to a function and then the function stores the result in them.

In the next major, we should consider returning `{ pattern: string, flags: string }` instead.